arches_controlled_lists/utils/skos.py

```diff
@@ -0,0 +1,204 @@
+                                ),
+                                "type": "broader",
+                                "target": list_item,
+                            }
+                        )
+                    elif predicate == SKOS.narrower:
+                        self.relations.append(
+                            {
+                                "source": list_item,
+                                "type": "narrower",
+                                "target": self.generate_uuidv5_from_subject(
+                                    baseuuid, object
+                                ),
+                            }
+                        )
+
+                    elif predicate == ARCHES.sortOrder:
+                        sortorder = int(self.unwrapJsonLiteral(str(object))["value"])
+
+                list_item.uri = uri
+                list_item.sortorder = sortorder
+                self.list_items[list_item_id] = list_item
+
+            ### Relationships ###
+            for relation in self.relations:
+                source = relation["source"]
+                target = relation["target"]
+                type = relation["type"]
+                if type == "narrower":
+                    self.list_items[target].parent = source
+                elif type == "broader":
+                    self.list_items[source].parent = target
+
+            with transaction.atomic():
+                List.objects.bulk_create(self.lists.values())
+                ListItem.objects.bulk_create(self.list_items.values())
+                ListItemValue.objects.bulk_create(self.list_item_values)
+
+                # TODO: Handle list item pk collisions in polyhierarhcies
+
+                ### Sort order ###
+                new_list_items = ListItem.objects.filter(
+                    list_id__in=[new_list.pk for new_list in self.lists.values()]
+                ).prefetch_related("children")
+
+                parents = set(
+                    chain(
+                        new_list_items.filter(parent__isnull=True),  # root items
+                        new_list_items.exclude(children__isnull=True),  # with children
+                    )
+                )
+                list_items_with_sortorder = []
+                for parent in parents:
+                    children = {}
+                    for child in parent.children.all().prefetch_related(
+                        "list_item_values"
+                    ):
+                        try:
+                            prefLabel = child.list_item_values.get(
+                                valuetype="prefLabel", language=default_lang
+                            )
+                        except ListItemValue.DoesNotExist:
+                            prefLabel = child.list_item_values.filter(
+                                valuetype="prefLabel"
+                            ).first()
+                        children[prefLabel.value] = child
```

Comment:
- I wouldn't do `.get()` and `first()`, I'd just do first()
- but either of these do new queries, so the prior prefetch isn't used

Like you said at standup we don't have to over-optimize here, but if you want -- you can provide a custom prefetch above to just get preflabels (or even the specific single preflabel you want with a case/when)